File: Semestrul 4/MPP2/Client-Server_2Limbaje/Problema8SC/Problema8SC/Problema8Model/src/main/java/problema8/model/ChildDetails.java
package problema8.model;

public class ChildDetails extends Entity<Long> {
    public String Name;
    public int Age;
    public long NumberOfEvents;

    public void setName(String name) {
        this.Name = name;
    }
    public String getName() {
        return Name;
    }

    public void setAge(int age) {
        this.Age = age;
    }
    public int getAge() {
        return Age;
    }

    public void setNumberOfEvents(long numberOfEvents) {
        this.NumberOfEvents = numberOfEvents;
    }
    public long getNumberOfEvents() {
        return NumberOfEvents;
    }

    public ChildDetails(Long id, String name, int age, long numberOfEvents) {
        setId(id);
        this.Name = name;
        this.Age = age;
        this.NumberOfEvents = numberOfEvents;
    }
}

File: Semestrul 4/MPP2/Problema8_Csharp_ORM/Model/ChildDetails.cs
namespace Problema8SC_CSharp.Model;

public class ChildDetails: Entity<long>
{
    public string Name { get; set; }
    public int Age { get; set; }
    public long NumberOfEvents { get; set; }
    
    public long Id
    {
        get => GetId();
        set => SetId(value);
    }
    
    public ChildDetails(long id, string name, int age, long numberOfEvents)
    {
        Id = id;
        Name = name;
        Age = age;
        NumberOfEvents = numberOfEvents;
    }
}